app/models/job.rb

```diff
@@ -102,7 +102,7 @@ def finished?
   end
 
   def queued?
-    JobQueue.queued?(id)
+    JobQueue.queued?(id) ||  DeployQueue.queued?(id)
```

Comment:
race condition ... could be just moved from queued into JobQueue and it would not be marked as queued ?

Reply:
Hm, how would I fix that? Have some sort of global lock on both queues when this method is called?

Reply:
- possibly handling queueing in job-queue
 - reuse job-queue lock